internal/services/amlfilesystem/aml_file_system_resource.go

```diff
@@ -0,0 +1,512 @@
+				Schema: map[string]*pluginsdk.Schema{
+					"key_url": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: keyVaultValidate.NestedItemId,
+					},
+
+					"source_vault_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: commonids.ValidateKeyVaultID,
+					},
+				},
+			},
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r AMLFileSystemResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r AMLFileSystemResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model AMLFileSystemModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AMLFileSystem.AmlFilesystems
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			id := amlfilesystems.NewAmlFilesystemID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for existing %s: %+v", id, err)
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			identity, err := expandAMLFileSystemIdentity(model.Identity)
+			if err != nil {
+				return err
+			}
+
+			properties := &amlfilesystems.AmlFilesystem{
+				Location: location.Normalize(model.Location),
+				Identity: identity,
+				Properties: &amlfilesystems.AmlFilesystemProperties{
+					Hsm:                expandAMLFileSystemHsmSetting(model.HsmSetting),
+					EncryptionSettings: expandAMLFileSystemKeyEncryptionKey(model.KeyEncryptionKey),
+					MaintenanceWindow:  expandAMLFileSystemMaintenanceWindowForCreate(model.MaintenanceWindow),
+					FilesystemSubnet:   model.SubnetId,
+					StorageCapacityTiB: float64(model.StorageCapacityInTb),
+				},
+				Sku: &amlfilesystems.SkuName{
+					Name: pointer.To(model.SkuName),
+				},
+				Zones: pointer.To(model.Zones),
+				Tags:  pointer.To(model.Tags),
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, *properties); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r AMLFileSystemResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.AMLFileSystem.AmlFilesystems
+
+			id, err := amlfilesystems.ParseAmlFilesystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model AMLFileSystemModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			properties := amlfilesystems.AmlFilesystemUpdate{
```

Comment:
Do we have a specific reason not to follow the best practice for [full update](https://github.com/hashicorp/terraform-provider-azurerm/blob/22c6d8c1494c7301bcd02382ba55dd4cb9644c64/contributing/topics/best-practices.md#:~:text=A%20full%20update%20would%20retrieve%20the%20existing%20object%20from%20the%20API%20and%20then%20patch%20it%2C%20for%20example%3A)?

Reply:
The type of model returned by GET API is different with the type of payload that UpdateThenPoll func requires. So I didn't follow that practice.

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/1d4d10ea-c93e-407b-8b50-2561a1b3e5e9)

![image](https://github.com/hashicorp/terraform-provider-azurerm/assets/19754191/188d497e-a21d-4468-ae42-b76b911486fc)

Reply:
Can't we manually assign the create parameter with the get response type?